code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -105,68 +109,48 @@ export function composeStory<TRenderer extends Renderer = Renderer, TArgs extend
     abortSignal: new AbortController().signal,
     step: (label, play) => story.runStep(label, play, context),
     canvasElement: globalThis?.document?.body,
-    context: null!,
     canvas: {},
     ...story,
+    context: null!,
+    mount: null!,
   };
 
   context.context = context;
+  context.mount = story.mount(context);
+  context.renderToCanvas = async () => {
+    const unmount = await story.renderToCanvas?.(
+      { storyContext: context } as RenderContext<TRenderer>,
+      context.canvasElement
+    );
+    if (unmount) {
+      cleanups.push(unmount);
+    }
+  };
 
-  const playFunction = story.playFunction
-    ? async (extraContext?: Partial<StoryContext<TRenderer, Partial<TArgs>>>) => {
-        Object.assign(context, extraContext);
-        return story.playFunction!(context);
-      }
-    : undefined;
-
-  let previousCleanupsDone = false;
+  const playFunction = (extraContext?: Partial<StoryContext<TRenderer, Partial<TArgs>>>) => {
+    Object.assign(context, extraContext);
+    return playStory(story, context);
+  };
 
   const composedStory: ComposedStoryFn<TRenderer, Partial<TArgs>> = Object.assign(
     function storyFn(extraArgs?: Partial<TArgs>) {
       context.args = {
         ...context.initialArgs,
         ...extraArgs,
       };
-
-      if (cleanups.length > 0 && !previousCleanupsDone) {
-        let humanReadableIdentifier = storyName;
-        if (story.title !== DEFAULT_STORY_TITLE) {
-          // prefix with title unless it's the generic ComposedStory title
-          humanReadableIdentifier = `${story.title} - ${humanReadableIdentifier}`;
-        }
-        if (storyName === DEFAULT_STORY_NAME && Object.keys(context.args).length > 0) {
-          // suffix with args if it's an unnamed story and there are args
-          humanReadableIdentifier = `${humanReadableIdentifier} (${Object.keys(context.args).join(
-            ', '
-          )})`;
-        }
-        console.warn(
-          dedent`Some stories were not cleaned up before rendering '${humanReadableIdentifier}'.
-          
-          You should load the story with \`await Story.load()\` before rendering it.`
```

Comment:
we are deprecating `Story.load`, and if you use `Story.play`, we do an auto clean up at the start of play

Reply:
There were test changes, mostly test being removed as they were not needed
https://github.com/storybookjs/storybook/pull/28383/files#diff-8ff993d697cbd953e3210bb683ac91e32318b65b43219970d00d0ee6cd8d19fbL262